pages/extensions/amp_example_preview/util/markdown_processor.py

```diff
@@ -0,0 +1,73 @@
+import os
+from grow.documents import document
+from .example_extractor import *
+from .preview import ExamplePreview
+from .constants import *
+
+import os
+
+EXAMPLE_TRIGGER = '[example'
+
+
+def trigger(doc, original_body, content):
+    if EXAMPLE_TRIGGER in original_body:
+        return _transform(doc, content)
+    return content
+
+
+def _transform(doc, content):
+
+    source_extractor = SourceCodeExtractor()
+    example_matches = source_extractor.find_examples_in_markdown(content)
+
+    all_imports = set()
+    all_templates = set()
+
+    pos = 0
+    count = 0
+    result = ''
+    for match in example_matches:
+
+        inline_example = match.inlineExample
+
+        count = count + 1
+
+        doc.pod.logger.debug('Found example {} in {}'.format(inline_example.index, doc.pod_path))
+
+        result = result + content[pos:match.startTagStart]
+
+        preview = ExamplePreview(index=inline_example.index,
+                                 mode=inline_example.preview,
+                                 playground=inline_example.playground,
+                                 source=inline_example.source)
```

Comment:
Sorry for not being clear enough. What I meant is to encapsulate options into an object:

```
opts = {
 
}
```

which is then passed around `ExamplePreview(opts)`. This avoids changing the ExamplePreview's constructor whenever adding a new option.

Reply:
The current solution provides those benefits. What would be the benefit of using a dictionary instead of the named varargs?

Reply:
all good - missed the python magic